batch/Makefile

```diff
@@ -1,30 +1,27 @@
 .PHONY: build build-test \
   push push-test \
   run-docker run \
-  test test-local deploy clean
+  test test-local deploy clean check
 
 PROJECT = $(shell gcloud config get-value project)
 PY_FILES = $(shell find batch -iname \*.py -not -exec git check-ignore -q {} \; -print)
-PY_CHECKERS = build/flake8 build/pylint
+PY_CHECKERS = flake8-stmp pylint-stmp
 BATCH_PORT ?= 5000
 
-clean:
-	-rm -r build
-
-build/conda-env: environment.yml
-	conda env update -f environment.yml || conda env create -f environment.yml
-	mkdir -p build && touch $@
+flake8-stmp: $(PY_FILES)
+	python3 -m flake8 batch
+	touch $@
 
-build/flake8: $(PY_FILES) build/conda-env
-	. ../loadconda && conda activate hail-batch && python -m flake8 batch
-	mkdir -p build && touch $@
+pylint-stmp: $(PY_FILES)
+	python3 -m pylint --rcfile batch/pylintrc batch --score=n
+	touch $@
 
-build/pylint: $(PY_FILES) build/conda-env
-	. ../loadconda && conda activate hail-batch && python -m pylint --rcfile batch/pylintrc batch --score=n
-	mkdir -p build && touch $@
+check: $(PY_CHECKERS)
 
 build:
-	docker build -t batch .
+	make -C ../docker build
+	-docker pull gcr.io/$(PROJECT)/batch
+	docker build -t batch --cache-from batch,base,ubuntu:18.04 .
```

Comment:
We should also push the `latest` tag on every deploy.

In general, I prefer explicit tags in image names because it's non-obvious that `docker pull gcr.io/$(PROJECT)/batch` pulls the `latest` tag. Even with that knowledge, `latest` is a misleadingly named tag. It really ought to be called `default`, it has no relation to the most recently pushed or most recently built image.

Reply:
And to reiterate: in current docker, this means if you change layer `k`, successfully build layers `k` to `k'`, and fail on layer `k'+1`, you will have to rebuild layers `k` to `k'` because your `--cache-from` does not include the anonymous, temporary images for `k` through `k'`.

Reply:
Agreed, explicit is better, I'll put latest everywhere.

Yes.  As far as I can tell, there's no way to cache from partial failed local builds and a pre-existing remote build.  Which sucks balls when you're trying to iterate on a big Dockerfile like pr-builder.